ktor-server/ktor-server-plugins/ktor-server-sse/common/test/io/ktor/server/sse/ServerSentEventsTest.kt

```diff
@@ -192,6 +195,115 @@ class ServerSentEventsTest {
         assertEquals(expectedOneEventData.lines(), actualOneEventData.toString().lines())
     }
 
+    class Person1(val age: Int)
+    class Person2(val number: Int)
+
+    @Test
+    fun testSerializerInRoute() = testApplication {
+        install(SSE)
+        routing {
+            sse(
+                "/person",
+                object : Serializer {
+                    override fun serialize(typeInfo: TypeInfo, data: Any): String {
+                        return when (typeInfo.type) {
+                            Person1::class -> {
+                                "Age ${(data as Person1).age}"
+                            }
+
+                            else -> {
+                                data.toString()
+                            }
+                        }
+                    }
+                }
+            ) {
+                repeat(10) {
+                    sendSerialized(Person1(it))
+                }
+            }
+        }
+
+        val client = createSseClient()
+
+        client.sse("/person") {
+            incoming.collectIndexed { i, person ->
+                assertEquals("Age $i", person.data)
+            }
+        }
+    }
+
+    @Test
+    fun testDifferentSerializers() = testApplication {
+        install(SSE)
+        routing {
+            sse(object : Serializer {
+                override fun serialize(typeInfo: TypeInfo, data: Any): String {
+                    return when (typeInfo.type) {
+                        Person1::class -> {
+                            "Age ${(data as Person1).age}"
+                        }
+
+                        Person2::class -> {
+                            "Number ${(data as Person2).number}"
+                        }
+
+                        else -> {
+                            data.toString()
+                        }
+                    }
+                }
+            }) {
+                sendSerialized(Person1(22))
+                sendSerialized(Person2(123456))
+            }
+        }
```

Comment:
I'd be interested in seeing an example with kotlinx-serialization-json, since this is probably the most common use-case for this feature.  We ought to shape the API so that this is easy to implement.

Reply:
Here is used kotlinx-serialization-json https://github.com/ktorio/ktor/blob/9ce8ec50db95e125e1c0aff644683fff89a242e1/ktor-server/ktor-server-plugins/ktor-server-sse/common/test/io/ktor/server/sse/ServerSentEventsTest.kt#L291